client/src/main/scala/geotrellis/admin/client/components/modal/LayerList.scala

```diff
@@ -16,71 +18,55 @@ import japgolly.scalajs.react.vdom.prefix_<^._
+  class LayerBackend($: BackendScope[ModelProxy[LayerModel], Unit]) {
 
-    def render(props: ModelProxy[LayerModel]) = {
-      <.div(
-        <.h3("Layer"),
-        props().layers.renderPending(_ > 500, _ => <.p("Loading...")),
-        props().layers.renderFailed(ex => <.p("Failed to load")),
-        props().layers.render(layers => {
-          <.div(
-            LayerForm(
-              LayerForm.Props(
-                options = layers.sortBy(_.name),
-                onSelect = layer => props.dispatch(SelectLayer(layer))
-              )
-            )
-          )
-        })
-      )
-    }
-  }
-
-  private val layerViewer = ReactComponentB[ModelProxy[LayerModel]]("LayerViewer")
-    .renderBackend[Backend]
-    .componentDidMount(scope => scope.backend.onMount(scope.props))
-    .build
-
-  def apply(props: ModelProxy[LayerModel]) = layerViewer(props)
-}
+    def onMount(proxy: ModelProxy[LayerModel]) =
+      Callback.when(proxy().layers.isEmpty)(proxy.dispatch(RefreshLayers))
 
-object LayerForm {
-
-  case class Props(options: Seq[LayerDescription], onSelect: Option[LayerDescription] => Callback)
-  case class State(value: js.UndefOr[ReactNode] = js.undefined)
-
-  class Backend($: BackendScope[Props, State]) {
-
-    def onChange(value: ReactNode) = {
+    def onChange(options: Seq[LayerDescription], value: ReactNode): Callback = {
       val selection = JSON.parse(JSON.stringify(value))
       val layerIndex: Int = selection.value.asInstanceOf[Int]
 
-      $.modState(_.copy(value = value)) >>
-        $.props >>= { (props: Props) => props.onSelect(Try(props.options(layerIndex)).toOption) }
+      $.props >>= { proxy: ModelProxy[LayerModel] =>
+        proxy.dispatch(SelectLayer(Try(options(layerIndex)).toOption))
+      }
     }
 
-    def render(state: State, props: Props) = {
-      val options =
-        props.options.zipWithIndex.map { case (option, index) =>
-          ValueOption[ReactNode](value = index, label = option.name)
-        }.toJSArray
-
-      Select(
-        options = options,
-        value = state.value,
-        onChange = onChange _,
-        placeholder = "Select a layer to view...".asInstanceOf[ReactNode]
-      )()
+    def render(proxy: ModelProxy[LayerModel]) = {
+      <.div(
+        <.h3("Layer"),
+        proxy().layers.renderEmpty(<.p("Fetching layers...")),
+        proxy().layers.renderFailed(ex => <.p("Failed to load")),
+        proxy().layers.render(layers => {
+          <.div({
+            val layDs: Seq[LayerDescription] = layers.sortBy(_.name)
+
+            val options: Seq[ValueOption[ReactNode]] = layDs.zipWithIndex.map { case (option, index) =>
+              ValueOption[ReactNode](value = index, label = option.name)
+            }
+
+            // For some reason, `.getOrElse` won't compile here.
+            val sel: JsCol[ReactNode] = proxy().selection.flatMap(ld =>
+              options.filter(_.label == ld.name).headOption.map(_.value)  // Naive.
```

Comment:
In Haskell, the default List type is in a Scala sense a lazy, immutable `Stream`. It's a typical idiom to do `head . filter`, knowing that the filter will stop as soon as a `head` is found. Would a Scala `Stream` do that?

The `Seq`s here are actually `Array`. Perhaps the total layer lists aren't long enough to worry about optimizing this.